sync-todo/v2/client/cpp/main.cpp

```diff
@@ -0,0 +1,269 @@
+#include "screens/authentication.hpp"
+#include "screens/error-modal.hpp"
+#include "screens/options.hpp"
+#include "screens/scroller.hpp"
+
+static ItemManager itemManager;
+
+int main() {
+    /* Read the contents of the atlasConfig.json to get the metadata for the App Services App.
+     * This path assumes you are running the app from a `/build` directory within this project. If you're
+     * running the app somewhere else, update the path to your atlasConfig.json accordingly. */
+    std::ifstream f("../atlasConfig.json");
+    nlohmann::json data = nlohmann::json::parse(f);
+    auto appConfigMetadata = data.template get<AppConfigMetadata>();
+    f.close();
+
+    // Refer to `AppState` for descriptions of these fields.
+    auto appState = AppState {
+      .newTaskSummary = "",
+      .newTaskIsComplete = false,
+      .errorMessage = "",
+      .hideCompletedTasks = false,
+      .subscriptionSelection = SubscriptionSelection::allItems,
+      .offlineModeSelection = OfflineModeSelection::offlineModeDisabled,
+      .customLoopCount = 0,
+      .frameCount = 0,
+      .eventCount = 0,
+      .screenDisplaying = DisplayScreen::placeholderComponent,
+      .screen = ftxui::ScreenInteractive::FitComponent()
+    };
+
+    auto appConfig = realm::App::configuration {
+            .app_id = appConfigMetadata.appId
+    };
+    auto app = std::make_shared<realm::App>(appConfig);
+    auto authManager =
+            std::make_shared<AuthManager>(app);
+
+    auto errorModal = ErrorModal().init(&appState);
+    auto authModal = Authentication().init(authManager);
+
+    // Populate a notLoggedInScreen when there is no logged-in user. The authentication modal will pop up
+    // over this notLoggedInScreen.
+    auto notLoggedInScreen =
+            ftxui::vbox({
+                                ftxui::text("Welcome to the Atlas Device SDK C++ Task Tracker"),
+                                ftxui::text("Please log in or register a user to proceed.")
+                        });
+
+    notLoggedInScreen = notLoggedInScreen | size(ftxui::WIDTH, ftxui::GREATER_THAN, 80);
+
+    auto notLoggedInContainer = ftxui::Container::Vertical({});
+    auto notLoggedInRenderer = Renderer(notLoggedInContainer, [=] {
+        auto content = ftxui::vbox({
+                                           notLoggedInScreen,
+                                   });
+        return window(ftxui::text(L" Todo Tracker "), content);
+    });
+
+    auto dashboardContainer = ftxui::Container::Vertical({
+
+                                                         });
+
+    auto dashboardRenderer = Renderer(dashboardContainer, [=] {
+        auto content = ftxui::vbox({
+                                           notLoggedInScreen,
+                                   });
+        return window(ftxui::text(L" Todo Tracker "), content);
+    });
+
+    // If there is a logged-in user, render the scrollable task list.
+    auto currentUser = app->get_current_user();
```

Comment:
When registering a user for the first time, I have to quit the app and re-start it to trigger this logic (because there is no currentUser when the app executes the first time).

Is the best solution for this to move this line up above the auth manager, pass a pointer to current user to auth manager, and set it once there is a logged-in user? Or is there a better way to get a user we can work with here in main after registering one?